presto-main/src/main/java/com/facebook/presto/operator/scalar/MapToMapCast.java

```diff
@@ -274,4 +283,15 @@ public static Block mapCast(
         mapBlockBuilder.closeEntry();
         return (Block) toMapType.getObject(mapBlockBuilder, mapBlockBuilder.getPositionCount() - 1);
     }
+
+    public static boolean canSkipDuplicateKeysCheck(Type castFromType, Type castToType)
+    {
+        if (castFromType.equals(INTEGER)) {
+            return castToType.equals(BIGINT) || castToType.equals(VARCHAR);
+        }
+        else if (castFromType.equals(BIGINT)) {
```

Comment:
nit: else is redundant here

Reply:
The fromType can be types other than integer and bigint, for these cases we currently do not consider for skipping duplicate check